module_renamer/commands/rename_imports.py

```diff
@@ -0,0 +1,91 @@
+from concurrent import futures
+
+import pasta
+import six
+from click._unicodefun import click
+from pasta.augment import rename
+from tqdm import tqdm
+
+from module_renamer.commands.utils import walk_on_py_files
+
+
+def rename_modules(project_path, path_to_moved_imports_file):
+    for path in project_path:
+        execute_rename(path, path_to_moved_imports_file)
+
+
+def execute_rename(project_path, path_to_moved_imports_file):
+    """
+    Main loop that interacts over all python files from the project and delegate
+    to an executor to parse each file
+
+    :param str project_path:
+        Path to the project that is going to be parsed.
+
+    :param str path_to_moved_imports_file:
+        Path of the file with the list of changed imports.
+    """
+    list_of_py_files = [i for i in walk_on_py_files(project_path)]
+    file_counter = len(list_of_py_files)
+
+    with futures.ThreadPoolExecutor(max_workers=30) as executor:
+        future_map = {
+            executor.submit(rename_file, py_file, path_to_moved_imports_file): py_file
+            for py_file in list_of_py_files
+        }
+
+        done_list = tqdm(futures.as_completed(future_map), total=file_counter, unit="files", leave=False)
+        for future in done_list:
+            try:
+                future.result()
+            except Exception as exc:
+                click.ClickException(exc)
```

Comment:
Probably a better way is to create a list of errors and display them at the end as a summary of problems. Also, `click.ClickException(exc)` doesn't do anything, it only instantiates a `ClickException` and throws the result away, you probably meant `raise click.ClickException`. Also [`ClickException` accepts a `message` parameter](http://click.pocoo.org/5/api/#click.ClickException), not an exception object.

More info: http://click.pocoo.org/5/exceptions/

Reply:
- [x]  1) Create a list of errors and display them
- [x]  2) Raise the exception... :grin:

1) The list of errors is going to be displayed as below
```
Error: The following exception(s) has occurred while parsing the files:                                                                                                          

File /tmp/tests/file_a (copy).py generated an exception: invalid syntax (<unknown>, line 1)
File /tmp/tests/file_a.py generated an exception: invalid syntax (<unknown>, line 1)
File /tmp/tests/file_a (another copy).py generated an exception: invalid syntax (<unknown>, line 1)
```

2) I added a new test to check if the exception is being thrown ;)